slapo/model_schedule/__init__.py

```diff
@@ -0,0 +1,21 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+"""Model schedule."""
+
+import os
+from os.path import abspath, dirname, join, isfile
+from inspect import getsourcefile
+from importlib import import_module
+
+from .api import apply_schedule
+
+# register all model schedules in the current folder
+path = dirname(abspath(getsourcefile(lambda: 0)))
+files = [f for f in os.listdir(path) if isfile(join(path, f)) and f != "__init__.py"]
+for file in files:
+    mod = import_module(f".{file.split('.')[0]}", package="slapo.model_schedule")
+    # register the schedule method using the decorator
+    try:
+        getattr(mod, "apply_schedule")
+    except Exception:
+        pass
```

Comment:
1. Use `hasattr` to avoid this unnecessary try-catch.

2. The schedule method has the same name as the dispatching API (apply_schedule). This seems a bit confusing, especially they have different function signature (the API has a key). Maybe using "_apply_schedule"?